control-plane/agents/src/bin/core/controller/registry.rs

```diff
@@ -130,19 +131,25 @@ impl Registry {
                 reconcile_period,
                 reconcile_idle_period,
                 reconciler: ReconcilerControl::new(),
-                config: Self::get_config_or_panic(&mut store, legacy_prefix_present).await,
+                config: parking_lot::RwLock::new(
+                    Self::get_config_or_panic(&mut store, legacy_prefix_present).await,
+                ),
                 max_rebuilds,
                 host_acl,
                 legacy_prefix_present,
             }),
         };
         registry.init().await;
 
-        // Remove all entries of v1 key prefix.
-        store
-            .delete_values_prefix(&product_v1_key_prefix())
-            .await
-            .expect("Product v1 key prefix deletion should not fail");
+        // Disable v1 compat if nexus_info keys are migrated.
+        if !registry.nexus_info_v1_present().await {
+            // Delete the v1 nexus_info keys by brute force.
+            delete_all_v1_nexus_info(&mut store)
+                .await
+                .expect("v1 nexus_info deletion should not fail");
+            // Disable the v1 compat mode.
+            registry.set_config(Self::disable_v1_compat(&mut store).await);
```

Comment:
can we move this entire block to a separate function and have a single catch all for the Result::Err instead of separate panics?
And the panic message should be something related to persistent store I guess?